src/src/components/Contexts/SettingsContext.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { ajax } from '../../utils/ajax';
+import { AuthContext } from './AuthContext';
+
+type UserSettings = {
+  readonly weekStart: number;
+};
+
+export function SettingsProvider({
+  children,
+}: {
+  readonly children: React.ReactNode;
+}): JSX.Element {
+  const authContext = React.useContext(AuthContext);
+  const [weekStart, setWeekStart] = React.useState<number>(0);
+  const [found, setFound] = React.useState<boolean>(false);
+
+  // If the token has been set, try an update
+  // Also make sure we havent already fetched
+  if (authContext.token && !found) {
+    ajax(
+      'https://www.googleapis.com/calendar/v3/users/me/settings/weekStart'
+    ).then((res) => {
+      res.json().then((val) => {
+        if (typeof val['value'] === 'string') {
+          let wstart = new Number(val['value']) as number;
+          setWeekStart(wstart);
+          setFound(true);
+        }
+      });
+    });
+  }
+
+  const userSettings = React.useMemo(
```

Comment:
Feature:
(either solve it here or open an issue so that we don't forget to solve it in the future)
Cache settings using `useStorage()` as they are not likely to change often. Can use the "stale while refresh strategy" - use the cached value while newer value is being retrieved

Reply:
Will create seperate issue as we want to get this merged sooner as it is blocking @jakeangus

Reply:
Covered by #201